packages/dx-react-grid-export/src/plugins/grid-exporter.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import {
+  Action, Plugin, PluginHost,
+} from '@devexpress/dx-react-core';
+import { GridExporterCore } from './grid-exporter-core';
+import { ExporterProps, ExporterState } from '../types/export';
+
+class GridExporterBase extends React.PureComponent<ExporterProps, ExporterState> {
+  state = {
+    isExporting: false,
+  };
+
+  exportGrid = () => {
+    this.setState({ isExporting: true });
+  }
+
+  finishExport = () => {
+    // NOTE: we can't update state during the render phase
+    window.setTimeout(() => {
+      this.setState({ isExporting: false });
+    }, 0);
+  }
+
+  render() {
+    const { isExporting } = this.state;
+    if (!isExporting) return null;
+
+    return (
+      <PluginHost>
+        <Plugin name="export">
+          <Action name="finishExport" action={this.finishExport} />
+        </Plugin>
+        <GridExporterCore {...this.props} />
+      </PluginHost>
```

Comment:
Why are we using our plugins here? I think it is excessively.

Reply:
Could you explain this in a greater detail?